llvm/lib/Target/AMDGPU/AMDGPUTargetMachine.cpp

```diff
@@ -734,21 +739,15 @@ void AMDGPUTargetMachine::registerPassBuilderCallbacks(PassBuilder &PB) {
         PM.addPass(createCGSCCToFunctionPassAdaptor(std::move(FPM)));
       });
 
-  // FIXME: Why is AMDGPUAttributor not in CGSCC?
-  PB.registerOptimizerLastEPCallback(
-      [this](ModulePassManager &MPM, OptimizationLevel Level) {
```

Comment:
Does this mean we never run amdgpuattributor without LTO?

Reply:
That is correct. For HIP (w/ or w/o rdc) and OpenMP, we always run LTO. Need to check OpenCL. If not, we probably want to run it as well.

Reply:
That is an issue. We still should be running this in non-LTO

Reply:
Then we will run it twice in LTO.